server/src/main/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfo.java

```diff
@@ -137,7 +139,14 @@ public static CompactionConfigValidationResult validatePartitionsSpecForMSQ(Part
       return CompactionConfigValidationResult.failure(
           "MSQ: Invalid partitioning type[%s]. Must be either 'dynamic' or 'range'",
           partitionsSpec.getClass().getSimpleName()
-
+      );
+    }
+    if (partitionsSpec instanceof DimensionRangePartitionsSpec
+        && ((DimensionRangePartitionsSpec) partitionsSpec).getTargetRowsPerSegment() == null) {
+      // Indicates maxRowsPerSegment is non-null. Cannot check directly for maxRowsPerSegment==null since a resolved
+      // value is computed from targetRowsPerSegment.
+      return CompactionConfigValidationResult.failure(
+          "MSQ: 'maxRowsPerSegment' not supported with 'range' partitioning"
```

Comment:
Should we instruct the user to remove this config from the compaction spec as well ?

Reply:
No longer required as both supported now.